### DISCUSSION TITLE
DTO

### DISCUSSION THREAD

**Discussion by Dnnbg123** [litestar-org/litestar#3713] [Q&A]
Hi! I'm studying the litestar documentation and I can't quite figure it out. What is recommended as the DTO used.

I've been testing [SQLAlchemyDTO](https://docs.litestar.dev/2/reference/plugins/sqlalchemy.html#litestar.plugins.sqlalchemy.SQLAlchemyDTO) However, I could not achieve validation of the string length. Using [PydanticDTO](https://docs.litestar.dev/2/reference/contrib/pydantic.html#litestar.contrib.pydantic.PydanticDTO) everything worked out fine.
Maybe I'm doing something wrong.
```python
class Base(DeclarativeBase): ...
class TestModel(Base):
    __tablename__ = "test"
    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(5))
    num: Mapped[int]
    bol: Mapped[bool]
    flo: Mapped[float]
    date: Mapped[date]

ReadDTO = SQLAlchemyDTO[TestModel]


class WriteDTO(SQLAlchemyDTO[TestModel]):
    config = DTOConfig(exclude={"id"})
```

Thanks to the authors for their work

**Comment by provinzkraut** [comment]
It's not clear from your question what you're trying to achieve. The DTOs are not interchangeable. `SQLAlchemyDTO` is to model SQLAlchemy models, while the `PydanticDTO` does the same for Pydantic. 

So to answer your question: The recommended DTO to use is the one appropriate for the data you are modelling.
